app.ts

```diff
@@ -0,0 +1,277 @@
+  content.innerHTML += table;
+
+  let rows: any = document.querySelector("#row-" + contentData[0] + ".rows");
+  for (let x = 0; x < contentData.length; x++) {
+    let rowCols = `<div class="row_cols">${contentData[x]}</div>`;
+    rows.innerHTML += rowCols;
+  }
+};
+
+// Clear Table Content
+
+const clearTable = () => {
+  const content: any = document.querySelector("#content");
+  const clear = "";
+
+  content.innerHTML = clear;
+};
+
+//// Fetch Requests
+
+// Heading Row (Getting the columns data)
+
+const getHeadings = () => {
+  try {
+    fetch("http://localhost:2050/columns", {
+      method: "GET",
+      headers: { "Content-Type": "application/json" },
+    })
+      .then((res) => res.text())
+      .then((data) => {
+        data = JSON.parse(data);
+        let headingData = data;
+        for (let i = 0; i < headingData.length; i++) {
+          createHeadingRow(headingData[i]);
+        }
+      });
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+// Table Content (Getting the table's data)
+
+const getTable = () => {
+  try {
+    fetch("http://localhost:2050/records?from=" + paramOne + "&to=" + paramTwo, {
+      method: "GET",
+      headers: { "Content-Type": "application/json" },
+    })
+      .then((res) => res.text())
+      .then((data) => {
+        data = JSON.parse(data);
+        let contentData = data;
+        for (let i = 0; i < contentData.length; i++) {
+          createTableContent(contentData[i]);
+        }
+      });
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+// Displays The Current Results Being Shown
+
+const stats = () => {
+  const pageStats: any = document.querySelector("#pageStats");
+
+  try {
+    fetch("http://localhost:2050/recordCount", {
+      method: "GET",
+      headers: { "Content-Type": "application/json" },
+    })
+      .then((res) => res.text())
+      .then((data) => {
+        data = JSON.parse(data);
+        let count = data;
+        let currentStats = "Showing results from " + paramOne + " to " + paramTwo + " out of " + count + " results.";
+        pageStats.innerHTML = currentStats;
+      });
+  } catch (error) {}
+};
+
+//// Debounce
+
+const debounce = (fn: any, delay: number) => {
+  let timer: number;
+  return function () {
+    clearTimeout(timer);
+    timer = setTimeout(() => {
+      fn();
+    }, delay);
+  };
+};
+
+//// Sizing And Resizing
+
+let resizing = () => {
+  let end = paramOne + getNoOfRows();
+
+  if (end > 999999) {
+    paramTwo = 999999;
+    paramOne = paramTwo - getNoOfRows();
+  } else {
+    paramOne;
+    paramTwo = paramOne + getNoOfRows();
+  }
+  clearTable();
+  getTable();
+  stats();
+};
+
+resizing = debounce(resizing, 500);
+
+window.addEventListener("resize", resizing);
+
+//// On Window Load
+
+window.onload = function () {
+  getHeadings();
+  getTable();
+  stats();
+};
+
+//// Navigation
+
+// Next
+const nextButton: any = document.querySelector("#next");
+let nextCount = 0;
```

Comment:
I'm struggling to understand why you need these count variables? They look like they are kinda contradicting the point of the debouncing?

Reply:
I'm using it to get the amount of times the button is being clicked so i can move next according to the amount